saiph/reduction/famd.py

```diff
@@ -401,11 +402,20 @@ def compute_categorical_cos2(
     -------
         dataframe of categorical cos2
     """
-    if model.U is not None and model.s is not None:
-        model_coords = pd.DataFrame(
-            model.U[:, :min_nf] * model.s[:min_nf],
-            columns=get_projected_column_names(min_nf),
-        )
+    # if categorical cos 2 is computed on a different df than the one used to fit the model
+    if model.U is not None and len(model.U) != len(df):
+        df_model = fit(df=df)
+        if df_model.U is not None and df_model.s is not None:
+            model_coords = pd.DataFrame(
+                df_model.U[:, :min_nf] * df_model.s[:min_nf],
+                columns=get_projected_column_names(min_nf),
+            )
+    else:
+        if model.U is not None and model.s is not None:
+            model_coords = pd.DataFrame(
+                model.U[:, :min_nf] * model.s[:min_nf],
+                columns=get_projected_column_names(min_nf),
+            )
```

Comment:
If we want to compute the cos2 of a df based on a model computed with another df, we need to use coords related to the df instead of using the one from the model